src/OpenTelemetry/Logs/LogRecord.cs

```diff
@@ -219,6 +221,24 @@ public void ForEachScope<TState>(Action<LogRecordScope, TState> callback, TState
             return ref this.Data;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal void ResetReferenceCount()
+        {
+            this.PoolReferenceCount = 1;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal void AddReference()
+        {
+            Interlocked.Increment(ref this.PoolReferenceCount);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal int RemoveReference()
```

Comment:
Minor, I personally found AddRef/Release commonly used, might be related to C++ only:

* https://www-archive.mozilla.org/projects/xpcom/book/cxc/html/appb#1151263
* https://www-archive.mozilla.org/projects/xpcom/book/cxc/html/appb#1125183
* https://docs.microsoft.com/en-us/windows/win32/api/unknwn/nf-unknwn-iunknown-addref

Reply:
How passionately do you feel about this? I looked around, don't see any good example or prior art in .NET. `AddRef` w/ `Release` seems a bit odd for me for a .NET API. `AddRef` and `RemoveRef` I think would be better?

Reply:
only 1% 😄

Reply:
Btw, this is a good article https://docs.microsoft.com/en-us/windows/win32/com/rules-for-managing-reference-counts